v0/destinations/bqstream/transform.js

```diff
@@ -0,0 +1,106 @@
+/* eslint-disable no-console */
+const { EventType } = require("../../../constants");
+const {
+  defaultRequestConfig,
+  defaultPostRequestConfig,
+  CustomError,
+  getSuccessRespEvents,
+  getErrorRespEvents,
+  isOAuthDestination
+} = require("../../util");
+
+/**
+ * This function puts the accessToken information in the transformed response
+ * to facilitate for a successfully authorised event
+ *
+ * @param {*} event - The event on which transformation is being performed
+ * @param {*} response - Transformation Response
+ * @returns Transformation Response bound with token information(may not be required as such)
+ */
+async function processAuth(event, response) {
+  // OAuth for BQStream destination
+  const { oauthAccessToken } = event.metadata;
+  if (!oauthAccessToken) {
+    throw new CustomError("Empty/Invalid access token", 500);
+  }
+  if (!response.headers) {
+    response.headers = {};
+  }
+  response.headers.Authorization = `Bearer ${oauthAccessToken}`;
+  return response;
+}
+
+const responseWrapper = response => {
+  const resp = defaultRequestConfig();
+  resp.endpoint = response.endPoint;
+  resp.method = defaultPostRequestConfig.requestMethod;
+  resp.headers = { ...response.headers, "Content-Type": "application/json" };
+  resp.body.JSON = response.payload;
+  return resp;
+};
+
+async function process(event) {
+  const { message } = event;
+  const { properties, type } = message;
```

Comment:
We can put a check for `properties` field, for cases where its a null type value or not event there. We can abort the event from transformer directly rather than still allowing the network request to go through.

Reply:
Makes sense. Added it